LeanCode.CodeAnalysis.ruleset

```diff
@@ -11,6 +11,32 @@
         <!-- NamingRule (interface with I) - we already have this rule in StyleCop -->
         <Rule Id="CA1816" Action="None"/>
         <!-- DisposeMethodsShouldCallSuppressFinalize - we don't use finalizers that much -->
+        <Rule Id="CA2007" Action="None"/>
+        <!-- DoNotDirectlyAwaitATaskAnalyzer - we don't need that -->
+        <Rule Id="CA1707" Action="None"/>
+        <!-- IdentifiersShouldNotContainUnderscores - disabling it only for tests is too painful -->
+        <Rule Id="CA1062" Action="None"/>
+        <!-- ValidateArgumentsOfPublicMethods - we rely on NRT -->
+        <Rule Id="CA1032" Action="None"/>
+        <!-- ImplementStandardExceptionConstructors - we deliberately ignore some constructors -->
+        <Rule Id="CA1716" Action="None"/>
+        <!-- IdentifiersShouldNotMatchKeywords - we really don't care about VB.NET or C++/CLI -->
+        <Rule Id="CA1002" Action="None"/>
+        <!-- DoNotExposeGenericLists - this rule doesn't affect us -->
+        <Rule Id="CA1030" Action="None"/>
+        <!-- UseEventsWhereAppropriate - we won't use events anyway -->
+        <Rule Id="CA1308" Action="None"/>
+        <!-- NormalizeStringsToUppercase - we know what we're doing -->
+        <Rule Id="CA1720" Action="None"/>
+        <!-- IdentifiersShouldNotContainTypeNames - we don't really care about it -->
+        <Rule Id="CA1019" Action="None"/>
+        <!-- DefineAccessorsForAttributeArguments - not really needed -->
+        <Rule Id="CA2234" Action="None"/>
+        <!-- PassSystemUriObjectsInsteadOfStrings - most of the time it will result in an exception -->
```

Comment:
Why 'most of the time it will result in an exception'? Anyway, I think it's a sane rule.

Reply:
When you use `HttpClient` with `BasePath`, and pass relative path to the calls, you can't use simple `new Uri("/mypath")` as it expects absolute paths. So simple changes there just result in wrong code.